service/src/main/java/uk/gov/hmcts/reform/fpl/service/GeneratedOrderService.java

```diff
@@ -96,11 +96,13 @@ public OrderTypeAndDocument buildOrderTypeAndDocument(OrderTypeAndDocument typeA
                 break;
             case SUPERVISION_ORDER:
                 orderBuilder.title(null);
-                ofNullable(orderMonths)
-                    .map(i -> time.now().plusMonths(orderMonths))
-                    .map(dateTime -> dateFormatterService.formatLocalDateTimeBaseUsingFormat(
-                        dateTime, "h:mma, d MMMM y"))
-                    .ifPresent(orderBuilder::expiryDate);
+                if (typeAndDocument.getSubtype() == FINAL) {
```

Comment:
Is the if required?
If the user selects interim then orderMonths would be null, as such the expiryDate wouldn't be added.

Reply:
oh right, yea then not needed 👍